src/pages/dataTableUtilities/reducer/getReducer.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+/**
+ * Method using a factory pattern variant to get a reducer
+ * for a particular page or page style.
+ *
+ * For a new page - create a constant object with the
+ * methods from reducerMethods which are composed to create
+ * a reducer. Add to PAGE_REDUCERS.
+ *
+ * Each key value pair in a reducer object should be in the form
+ * action type: reducer function, returning a new state object
+ *
+ */
+
+import {
+  filterData,
+  editTotalQuantity,
+  focusNextCell,
+  selectRow,
+  deselectRow,
+  deselectAll,
+  focusCell,
+  sortData,
+} from './reducerMethods';
+
+/**
+ * Used for actions that should be in all pages using a data table.
+ */
+const BASE_TABLE_PAGE_REDUCER = {
+  focusNextCell,
+  focusCell,
+  sortData,
+};
+
+const customerInvoice = {
+  ...BASE_TABLE_PAGE_REDUCER,
+  filterData,
+  editTotalQuantity,
+  selectRow,
+  deselectRow,
+  deselectAll,
+};
+
+const PAGE_REDUCERS = {
+  customerInvoice,
+};
```

Comment:
as above, I am trying to go for similar logic with both `reducer`, `columns` and ideally `PageInfo` (i.e. program requisitions have different `PageInfo` - I think it will be a common usecase in future customizations)

Reply:
Discuss above https://github.com/openmsupply/mobile/pull/1117#discussion_r315021137